Scripts/Runtime/Logging/.CSProject/UnityLogListener.cs

```diff
@@ -80,18 +86,21 @@ private void SendToLogger(UnityEngine.Object context, LogLevel logLevel, string
             // Check to see if this call went through Unity's new stubbed out logging class that
             // currently just proxies the existing logging system.
             // This was found in the v0.17 of the Unity.Entities package
-            if(stackFrame.GetMethod().ReflectedType.FullName == "Unity.Debug")
+            if (stackFrame.GetMethod()?.ReflectedType.FullName == "Unity.Debug")
             {
                 // Go one deeper to overcome the proxy.
                 stackFrame = new StackFrame(5, true);
             }
-            object loggerContext = ResolveLogContext(context, stackFrame);
-            Log.GetLogger(loggerContext).AtLevel(logLevel, message, stackFrame.GetFileName(), stackFrame.GetMethod().Name, stackFrame.GetFileLineNumber());
+
+            Log.Logger logger = context != null ? Log.GetLogger(context) : Log.GetStaticLogger(ResolveContextFromStack(stackFrame));
+            logger.AtLevel(logLevel, message, stackFrame.GetFileName(), stackFrame.GetMethod()?.Name, stackFrame.GetFileLineNumber());
         }
 
-        private object ResolveLogContext(UnityEngine.Object unityContext, StackFrame stackFrame)
+        private Type ResolveContextFromStack(StackFrame stackFrame)
         {
-            return unityContext == null ? stackFrame.GetMethod().ReflectedType : (object)unityContext;
+            MethodBase method = stackFrame.GetMethod();
+
+            return method != null ? method.ReflectedType : typeof(UnknownContext);
```

Comment:
Looks like it used to use `Log.GetLogger(...)` in all cases before, but now if the context is null it uses `Log.GetStaticLogger(...)`, what's the difference here?

Reply:
`Log.GetStaticLogger` vs `Log.GetLogger` both give you a `Log.Logger` instance but using different parameters. The idea is that most of the time you're in a non-static class and want to build a logger specific to that instance so you pass in `this`.

With static classes though when we build a logger we don't have an instance so we have to pass in the type.

Really, this just resolves to setting `Log.Logger.m_DerivedTypeName` with `insance.GetType().Name`(GetLogger) vs `type.Name`(GetStaticLogger)

---

In this specific case `GetStaticLogger` is used because we only have access to the calling type (at best) and not the calling instance.